app/models/events/new_discussion.rb

```diff
@@ -1,7 +1,8 @@
 class Events::NewDiscussion < Event
   def self.publish!(discussion)
     event = create!(kind: 'new_discussion',
-                    eventable: discussion)
+                    eventable: discussion,
+                    sequence_id: 0)
```

Comment:
So the logic for sequencing looks like this:
```
  acts_as_sequenced scope: :discussion_id, column: :sequence_id, skip:
lambda {|e| e.discussion.nil? }
```

but, if I make a 'new_discussion' event:
```
e.discussion_id # => nil
e.discussion # => the new discussion object (because we alias 'discussion'
to 'eventable' in new_discussion.rb)
```

This means it gets sequenced, with a scope of 'nil'
This means it gets put in the same pile with all the other new_discussion
events (maybe other events too, I dunno)
This means it has a really high sequence id, because it's outside the
appropriate channel

So that's the bug, but I'm hesitant to mess with anything too deep in the
event pile (such as making it so that 'new_discussion' events have a
discussion_id, all of a sudden), because I think it'll break stuff.

On Thu, Aug 6, 2015 at 5:19 PM, Robert Guthrie <notifications@github.com>
wrote:

> In app/models/events/new_discussion.rb
> <https://github.com/loomio/loomio/pull/2378#discussion_r36481248>:
>
> > @@ -1,7 +1,8 @@
> >  class Events::NewDiscussion < Event
> >    def self.publish!(discussion)
> >      event = create!(kind: 'new_discussion',
> > -                    eventable: discussion)
> > +                    eventable: discussion,
> > +                    sequence_id: 0)
>
> How does it get a sequenceId of 40,000? I'm still confused and happy to
> chat in person about it.. but I don't understand .. The general idea is
> that only events with sequence id count towards the discussion thread... if
> this does not have a sequence id, then how does it sometimes get a very
> high one?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2378/files#r36481248>.
>